test/unit/channel_state.js

```diff
@@ -110,8 +110,8 @@ describe('ChannelState addMessagesSorted', function () {
 		expect(state.messages).to.have.length(10);
 		state.addMessagesSorted([generateMsg({ id: 'id', date: `2020-01-01T00:00:00.007Z` })]);
 		expect(state.messages).to.have.length(11);
-		expect(state.messages[7].id).to.be.equal('7');
-		expect(state.messages[8].id).to.be.equal('id');
+		expect(state.messages[7].id).to.be.equal('id');
+		expect(state.messages[8].id).to.be.equal('7');
```

Comment:
One hurdle of the midpoint check addition is that message with the same `created_at` will be added before existing one but since the existence of the messages with the same `created_at` in real-world applications is *almost* impossible - I adjusted the test. 

Even without midpoint check, the previous solution - in case of multiple messages with the same `created_at` - the new message would be placed at most one index after the already existing one which is still incorrect.